api.planx.uk/modules/flows/validate/helpers.ts

```diff
@@ -39,11 +43,16 @@ export const numberOfComponentType = (
     (entry): entry is [string, Node] => isComponentType(entry, type),
   );
   if (fn) {
-    nodeIds
-      ?.filter(([_nodeId, nodeData]) => nodeData?.data?.fn === fn)
-      ?.map(([nodeId, _nodeData]) => nodeId);
+    if (type === ComponentType.Answer) {
+      return nodeIds
+        ?.filter(([_nodeId, nodeData]) => nodeData?.data?.val === fn)
+        ?.map(([nodeId, _nodeData]) => nodeId)?.length;
+    } else {
+      return nodeIds
+        ?.filter(([_nodeId, nodeData]) => nodeData?.data?.fn === fn)
+        ?.map(([nodeId, _nodeData]) => nodeId)?.length;
+    }
   } else {
-    nodeIds?.map(([nodeId, _nodeData]) => nodeId);
+    return nodeIds?.map(([nodeId, _nodeData]) => nodeId)?.length;
```

Comment:
These changes are unrelated to the most recent planx-core change, but came across them when investigating why the `validate.test.ts` tests were still passing without a mock in this module - spotted a data field issue & additionally added test coverage ! (Data field issue does _not_ impact current live usage of these helper functions, but likely would have cropped up next time we try to use elsewhere!)